### ISSUE TITLE
Feature request: Notes to record and save bookmarked codes

### ISSUE THREAD

**Issue by OpenNetSurfer** [korcankaraokcu/PINCE#252]
Newbie here....

Recently I have been using PINCE to debug wine games and I noticed that there are some features missing like in cheat engine. I used to use cheat engine long back and mainly the following feature seems to be quite helpful to be added:

- Notes to record what you found
- Save the addresses of machine code you found

**Comment by Bloodiko** [comment]
I can see a potential "Bookmark adresses" feature for the current process / in a cheattable or some other integration.

But writing notes could be simply done in an external txt file. Not sure if this should be added to pince at all. 
It feels like redundent functionality.

**Comment by korcankaraokcu** [comment]
> Notes to record what you found

You can already save tables with descriptions, can you elaborate further on this?

> Save the addresses of machine code you found

We already have bookmarks but yeah, they aren't saved into the file. I'll consider this

**Comment by OpenNetSurfer** [comment]
> > Notes to record what you found
> 
> You can already save tables with descriptions, can you elaborate further on this?

I simply meant a note feature like in cheat engine, where i can write in it what i found and save it  inside the PINCE save file. I wasn't talking about cheat tables.


>But writing notes could be simply done in an external txt file. Not sure if this should be added to pince at all.
It feels like redundent functionality.

I don't think it's a redundant feature. It sure does same some time instead of opening another text editor

**Comment by OpenNetSurfer** [comment]
> Currently Memory Adresses allow for comments. You can save adresses in your address table, set a comment on those, and save the whole table as a cheat table file.
>This will include all notes you have set on those addresses.

That's not helpful.

>Also, you can set bookmarks in the dissassembler, but they are not persistant.

Yes this is what I was suggesting.

**Comment by OpenNetSurfer** [comment]
Should the Notes (simple text editor in PINCE) feature be added as an dialog or a widget? It should be in a separate window.

**Comment by brkzlr** [comment]
This is being worked on by @Bloodiko, no need to make a PR for this.

**Comment by OpenNetSurfer** [comment]
Ok cool. Should I make the bookmarked disassembled codes also be saved?

**Comment by brkzlr** [comment]
No, like I said, this is being worked on by Ebiko and requires significant backend changes.

**Comment by OpenNetSurfer** [comment]
Understood
